server/middleware/auth.ts

```diff
@@ -20,7 +20,7 @@ export const attachUser: RequestHandler = async (req, res, next) => {
   next();
 };
 
-export const requireUser: RequestHandler = async (req, res, next) => {
+export const requireUser: AnyRequestHandler = async (req, res, next) => {
```

Comment:
#6 introduced a typescript error because request handlers in a chain are expected to be of the same type.  We want these to be generic because they'll be used with a myriad of different controllers, each with its own typed request/response/params/etc, and we don't really care about those in these middleware functions.